internal/webhooks/clusterclass_test.go

```diff
@@ -1318,6 +1320,824 @@ func TestClusterClassValidationWithClusterAwareChecks(t *testing.T) {
 	}
 }
 
+func TestClusterClassValidationWithVariableChecks(t *testing.T) {
+	// NOTE: ClusterTopology feature flag is disabled by default, thus preventing to create or update ClusterClasses.
+	// Enabling the feature flag temporarily for this test.
+	defer utilfeature.SetFeatureGateDuringTest(t, feature.Gates, feature.ClusterTopology, true)()
+
+	clusterClassBuilder := builder.ClusterClass(metav1.NamespaceDefault, "class1").
+		WithInfrastructureClusterTemplate(
+			builder.InfrastructureClusterTemplate(metav1.NamespaceDefault, "inf").Build()).
+		WithControlPlaneTemplate(
+			builder.ControlPlaneTemplate(metav1.NamespaceDefault, "cp1").
+				Build())
+	tests := []struct {
+		name            string
+		oldClusterClass *clusterv1.ClusterClass
+		newClusterClass *clusterv1.ClusterClass
+		clusters        []client.Object
+		expectErr       bool
+	}{
+		{
+			name: "Pass if a Minimum ClusterClassVariable is changed but does not invalidate an existing ClusterVariable",
+			clusters: []client.Object{
+				builder.Cluster(metav1.NamespaceDefault, "cluster1").
+					WithLabels(map[string]string{clusterv1.ClusterTopologyOwnedLabel: ""}).
+					WithTopology(
+						builder.ClusterTopology().
+							WithClass("class1").
+							WithVariables(
+								clusterv1.ClusterVariable{
+									Name: "cpu",
+									Value: apiextensionsv1.JSON{
+										Raw: []byte(`4`),
+									},
+								}).
+							Build()).
+					Build(),
+			},
+			oldClusterClass: clusterClassBuilder.
+				WithVariables(
+					clusterv1.ClusterClassVariable{
+						Name: "zone",
```

Comment:
```suggestion
						Name: "cpu",
```
same in l.1375

The test currently passes because there is no schema validation for the cpu variable.